### ISSUE TITLE
DEPR: DatetimeIndex/TimedeltaIndex constructor keywords

### ISSUE THREAD

**Issue by jbrockmendel** [pandas-dev/pandas#55499]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
NA
```

### Issue Description

`DatetimeIndex.__new__` has constructor keywords (excluding ones present in the base class and ones already deprecated) `freq`, `tz`, `ambiguous`, `dayfirst`, `yearfirst`.  `dayfirst` and `yearfirst` overlap with the `pd.to_datetime` keywords.

Could we deprecate some/all of these keywords and point users to `pd.to_datetime` instead?

- `freq` we could either exempt or make a post-construction thing by making `_with_freq` public.
- `to_datetime` has `utc` instead of `tz`.  I generally like `tz` better than `utc` as a keyword, but deprecating `utc` might be painful. We could add `tz` to `to_datetime` and disallow passing tz-and-utc.
- Adding `ambiguous` to to_datetime wouldn't be too bad aside from general keyword fatigue.

(similarly `TimedeltaIndex` has `unit` and `freq`, `unit` being present in `pd.to_timedelta`)

cc @MarcoGorelli @mroeschke @jorisvandenbossche 

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

**Comment by MarcoGorelli** [comment]
sounds good, would favour moving towards to_datetime as being _the_ way to parse into datetime

**Comment by mroeschke** [comment]
Agreed, would be simpler if DTI/TDI just accepted correctly typed arrays

**Comment by jbrockmendel** [comment]
> the reason that we went with utc=True/False in the past for to_datetime might be that a tz keyword can be a ambiguous on how to interpret the input data?

I've looked into this and determined the following:

In the DatetimeIndex constructor `ambiguous` is only used in cases where 1) the data is dt64-naive or 2) the data is object and array_to_datetime does not infer a tz.  In both of these cases, `DatetimeIndex(data, tz=tz, ambiguous=ambig)` is equivalent to `DatetimeIndex(data).tz_localize(tz, ambiguous=ambig)` and also to `pd.to_datetime(data).tz_localize(tz, ambiguous=ambig)`.

We _could_ push ambiguous into the cython `array_to_datetime_with_tz`, which would allow us to respect `ambiguous` in some mixed-type data cases, but those cases would be pretty contrived.  Moreover, we could only do this for a subset of the `ambiguous`s that we support, e.g. not "infer"` (allowing an arraylike would be possible but a PITA).

So i think we are safe:

1) deprecating the ambiguous keyword in DatetimeIndex, telling users to call .tz_localize after construction instead
2) Adding a `tz` keyword to to_datetime _without_ a `ambiguous` keyword (mutually exclusive with `utc`, which we could deprecated)
3) deprecating dayfirst, yearfirst in DatetimeIndex.

**Comment by jbrockmendel** [comment]
Tried deprecating ambiguous, dayfirst, yearfirst in DatetimeIndex and hit an unexpected snag: when a freq is passed, we currently pass ambiguous to _validate_frequency, which appears to need it in `test_resample_origin_with_day_freq_on_dst`.  Doesn't look insurmountable.
